### ISSUE TITLE
Generate classes even if there was a validation error

### ISSUE THREAD

**Issue by smaugho** [androidannotations/androidannotations#1976]
Right now AA skips the generation of all the Enhanced Classes  when there was a validation error in any place in the project. This causes a huge list of errors to be displayed and it is almost impossible to find the error that is causing the problem

 A good option, would be to generate all the classes, except the one causing the error... This would be the simplest solution. 

Even though, in most of the scenarios would be useful even generate the class with the validation error as described [here](https://github.com/androidannotations/androidannotations/issues/1868#issuecomment-269757610), but this would require probably a revision of every annotation handler in the framework.

**Comment by dodgex** [comment]
I'm currently investigating this. At the moment it looks to be a fairly easy task. I already got a build where it only produces the validation errors but generates classes (at least if it is not an error on a class level annotation). But I want to do some more tests. :)

**Comment by dodgex** [comment]
@smaugho could you checkout and build a local snapshot of https://github.com/androidannotations/androidannotations/pull/1977 and see how it behaves for your project?

**Comment by smaugho** [comment]
Yes I can, I'll be on this now...

**Comment by smaugho** [comment]
Well, I'm making some tests with it... but right now it is failing

![im](https://cloud.githubusercontent.com/assets/6622086/23921942/c5371952-0900-11e7-8150-a780c3b34846.png)

The error in the LogFile:

The error: ERROR:74 - Resource name not found in R.id: modalEditExpenseTEST

Please note that I don't have any only-AA here, I use DecleX, but this is similar to having a plugin applied no difference in behavior should be noted, all the features from the framework should keep the same (but better that you are ware for any comparison ;) )... I'm investigating here what happens after there's one invalidation, and if it gets to the generate method... I will let you know the results. 

This project I'm testing is not that complex, I have some complex one where I can test after, but well, it should pass this one..

**Comment by smaugho** [comment]
Ok, in general, it generated the classes, but for some reason they were not detected by "Java" I had this problem before, just trying to remember what was it :).. I will try with something simpler. 


```log
21:50:12.495 [Daemon worker Thread 3] INFO :340 - Number of files generated by AndroidAnnotations: 27
21:50:12.497 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.example.expenses.auth.Auth_
21:50:12.584 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.example.expenses.fragment.ExpenseDetailsFragmentActionHolder
21:50:12.585 [Daemon worker Thread 3] DEBUG:65 - Generating class with no originating element: com.dspot.declex.example.expenses.fragment.ExpenseDetailsFragmentActionHolder
21:50:12.598 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.example.expenses.fragment.ExpenseDetailsFragment_
21:50:12.711 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.example.expenses.fragment.ExpensesListFragmentActionHolder
21:50:12.712 [Daemon worker Thread 3] DEBUG:65 - Generating class with no originating element: com.dspot.declex.example.expenses.fragment.ExpensesListFragmentActionHolder
21:50:12.731 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.example.expenses.fragment.ExpensesListFragment_
21:50:12.887 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.example.expenses.fragment.LoginFragmentActionHolder
21:50:12.887 [Daemon worker Thread 3] DEBUG:65 - Generating class with no originating element: com.dspot.declex.example.expenses.fragment.LoginFragmentActionHolder
21:50:12.896 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.example.expenses.fragment.LoginFragment_
21:50:12.954 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.example.expenses.fragment.NewAccountFragmentActionHolder
21:50:12.955 [Daemon worker Thread 3] DEBUG:65 - Generating class with no originating element: com.dspot.declex.example.expenses.fragment.NewAccountFragmentActionHolder
21:50:12.973 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.example.expenses.fragment.NewAccountFragment_
21:50:13.30 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.example.expenses.fragment.ProfileFragmentActionHolder
21:50:13.30 [Daemon worker Thread 3] DEBUG:65 - Generating class with no originating element: com.dspot.declex.example.expenses.fragment.ProfileFragmentActionHolder
21:50:13.39 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.example.expenses.fragment.ProfileFragment_
21:50:13.115 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.example.expenses.fragment.StatisticsDetailsFragmentActionHolder
21:50:13.115 [Daemon worker Thread 3] DEBUG:65 - Generating class with no originating element: com.dspot.declex.example.expenses.fragment.StatisticsDetailsFragmentActionHolder
21:50:13.125 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.example.expenses.fragment.StatisticsDetailsFragment_
21:50:13.177 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.example.expenses.fragment.StatisticsFragmentActionHolder
21:50:13.177 [Daemon worker Thread 3] DEBUG:65 - Generating class with no originating element: com.dspot.declex.example.expenses.fragment.StatisticsFragmentActionHolder
21:50:13.195 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.example.expenses.fragment.StatisticsFragment_
21:50:13.239 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.example.expenses.model.Expense_
21:50:13.325 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.example.expenses.model.Token_
21:50:13.376 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.example.expenses.model.User_
21:50:13.459 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.example.expenses.model.WeekExpense_
21:50:13.515 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.example.expenses.ExpensesManagerApplication_
21:50:13.523 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.example.expenses.activity.LoginActivityActionHolder
21:50:13.524 [Daemon worker Thread 3] DEBUG:65 - Generating class with no originating element: com.dspot.declex.example.expenses.activity.LoginActivityActionHolder
21:50:13.541 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.example.expenses.activity.LoginActivity_
21:50:13.560 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.example.expenses.activity.MainActivityActionHolder
21:50:13.561 [Daemon worker Thread 3] DEBUG:65 - Generating class with no originating element: com.dspot.declex.example.expenses.activity.MainActivityActionHolder
21:50:13.570 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.example.expenses.activity.MainActivity_
21:50:13.644 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.event.BackPressedEvent
21:50:13.645 [Daemon worker Thread 3] DEBUG:65 - Generating class with no originating element: com.dspot.declex.event.BackPressedEvent
21:50:13.656 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.event.Logout
21:50:13.656 [Daemon worker Thread 3] DEBUG:65 - Generating class with no originating element: com.dspot.declex.event.Logout
21:50:13.666 [Daemon worker Thread 3] DEBUG:57 - Generating class: com.dspot.declex.event.UpdateUIEvent
21:50:13.666 [Daemon worker Thread 3] DEBUG:65 - Generating class with no originating element: com.dspot.declex.event.UpdateUIEvent
```

**Comment by WonderCsabo** [comment]
@dodgex i tried your PR. It is too eager now. 🤣  It generated the class itself where the error is placed. 

Moreover, a weird thing happened: the generated class for my `@SharedPref` exists, however Dagger cannot find it, it does not see that generated class. 😕

**Comment by smaugho** [comment]
Yes this is what is happening to me as well, basically it generates the class, so it exists, but the compiler some how doesn't see them, and generate this error... 

Maybe this is useful, in my case, I generate another class that should be processed in the next round of the annotation processor

```java
Note: Time measurements: [Whole Processing = 2207 ms], [Generate Sources = 1304 ms], [Process Annotations = 770 ms], [Validate Annotations = 92 ms], [Find R Classes = 12 ms], [Extract Annotations = 7 ms], [Extract Manifest = 5 ms], [Run ADI = 1 ms], 
Note: Finish processing
Round 2:
	input files: {}
	annotations: []
	last round: true
Note: Executing Declex
Note: Start processing for 0 annotations on 0 elements
Note: Time measurements: [Whole Processing = 0 ms], 
Note: Finish processing
100 errors
1 warning
```

But the annotation processing stops (and it flags this as the last round, this mean that from the generated class, where some of them have annotations, they were completely omitted).

**Comment by smaugho** [comment]
Ok got it.. I had this problem before... I'm no sure if it is the same but it is a clue.. if one of the generated classes has an error (so, they have something that is not valid in Java), then the annotation processor will stop exactly like this... I will check this

**Comment by WonderCsabo** [comment]
After looking it again, the compiler does not see any of the generated classes, but they are indeed present. So i am seeing the same error as @smaugho .

**Comment by smaugho** [comment]
Maybe useful, if I comment the LOGGER reports:

![im](https://cloud.githubusercontent.com/assets/6622086/23925589/368bd7e6-0910-11e7-9206-e6306fa8dd74.png)

Everything works like it should, so that specific Annotation where the validation failed, is omitted, and all the classes are generated and recognized.

**Comment by smaugho** [comment]
It works also if I change the LOGGER.error by LOGGER.warn... so definitely it is there ;)

**Comment by smaugho** [comment]
I solved it.. please check the solution, I made a pull request. 

Basically, the only way that I found to keep the compiler working after the first round, was to write the errors to the console and Processing Environment Messager when the annotation processing was over.

Please note that I create Statics lists to save the messages till the end of the process.. This is working very well here, but I'm not sure of what implications could have in a different compiler (maybe when used with Eclipse and ANT)..

**Comment by dodgex** [comment]
@runforprogram the changes are merged but after the 4.4 release, so the first release containing these changes will be 4.5 (currently there are only snapshots for this version)
